src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -0,0 +1,575 @@
+
+        self.ahs_program = ahs_program
+
+        return ahs_program
+
+    def generate_samples(self):
+        r"""Returns the computational basis samples measured for all wires.
+
+        Returns:
+             array[complex]: array of samples in the shape ``(dev.shots, dev.num_wires)``
+        """
+        return [self._result_to_sample_output(res) for res in self.samples.measurements]
+
+    def _validate_operations(self, operations: List[ParametrizedEvolution]):
+        """Confirms that the list of operations provided contains a single ParametrizedEvolution
+        from a HardwareHamiltonian with only a single, global pulse
+
+        Args:
+            operations(List[ParametrizedEvolution]): a list containing a single
+                ParametrizedEvolution operator
+        """
+
+        if len(operations) > 1:
+            raise NotImplementedError(
+                f"Support for multiple ParametrizedEvolution operators in a single circuit is "
+                f"not yet implemented. Received {len(operations)} operators."
+            )
+
+        ev_op = operations[0]  # only one!
+
+        if not isinstance(ev_op.H, HardwareHamiltonian):
+            raise RuntimeError(
+                f"Expected a HardwareHamiltonian instance for interfacing with the device, but "
+                f"recieved {type(ev_op.H)}."
+            )
+
+        if not set(ev_op.wires) == set(self.wires):
+            raise RuntimeError(
+                f"Device contains wires {self.wires}, but received a `ParametrizedEvolution` "
+                f"operator working on wires {ev_op.wires}. Device wires must match wires of "
+                f"the evolution."
+            )
+
+        if len(ev_op.H.settings.register) != len(self.wires):
+            raise RuntimeError(
+                f"The defined interaction term has register {ev_op.H.settings.register} of length "
+                f"{len(ev_op.H.settings.register)}, which does not match the number of wires on "
+                f"the device ({len(self.wires)})"
+            )
+
+    def _validate_pulses(self, pulses: List[HardwarePulse]):
+        """Confirms that the list of HardwarePulses describes a single, global pulse
+
+        Args:
+            pulses: List of HardwarePulses
+
+        Raises:
+            RuntimeError, NotImplementedError"""
+
+        if not pulses:
+            raise RuntimeError("No pulses found in the ParametrizedEvolution")
+
+        if len(pulses) > 1:
+            raise NotImplementedError(
+                f"Multiple pulses in a Hamiltonian are not currently supported. "
+                f"Received {len(pulses)} pulses."
+            )
+
+        if pulses[0].wires != self.wires:
+            raise NotImplementedError(
+                f"Only global drive is currently supported. Found drive defined for subset "
+                f"{[pulses[0].wires]} of all wires [{self.wires}]"
+            )
+
+    def _create_register(self, coordinates: List):
```

Comment:
Rather than a `List`, should this be a `Tuple[float, float]`? Same goes for docstring

Reply:
No, the format described in the doc-string is accurate. 

To elaborate, a user might pass [[0, 5], [5, 5], [10, 17]] in PennyLane. Then a register is created, and for each pair [x, y] in the list, the units are converted from um to m and then added to the register.

Reply:
Sorry, misread a line; should the List then be a List[Tuple[float, float]]?

Reply:
Sure! That would be more specific